packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewHostApi.m

```diff
@@ -251,7 +253,11 @@ - (void)loadHTMLForWebViewWithIdentifier:(nonnull NSNumber *)identifier
 
 - (void)reloadWebViewWithIdentifier:(nonnull NSNumber *)identifier
                               error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
-  [[self webViewForIdentifier:identifier] reload];
+  if ([self webViewForIdentifier:identifier].URL == NULL && _previousUrlRequest != NULL) {
```

Comment:
Why is the first request special here? If I successfully load page A, then try to load page B and it fails, then I reload, why should that behave differently from if I hadn't previously loaded A?

Reply:
The iOS WebView can not call reload if an initial loading page A fails. Could you please refer to following comment. https://github.com/flutter/plugins/pull/6887#issuecomment-1374397017

Reply:
That comment highlights the problem I am raising here, it doesn't address it. Your description of the behavior in that comment indicates that *all* reloads of failed loads behave differently on iOS and Android, but you are changing the behavior *only* if it's the first load.

You are special-casing the first load here, making the iOS behavior internally inconsistent (in addition to still inconsistent with Android).

Reply:
The difference in behavior between iOS and Android reloads is due to the difference in the behavior of the WebView on the platform side, so we think it is basically better not to unify them.
However, if reload is simply used, the https://github.com/flutter/flutter/issues/47677 problem will occur, so we think it is better to deal with this exceptionally. What do you think?

Reply:
You say it's better not to unify then, but then you are unifying them (and changing away from platform behavior) only for the first load. That leaves something that doesn't match across platforms, but also doesn't match platform behavior, and isn't internally consistent.

I think the next step here is a [design document](https://flutter.dev/go/template) describing the difference and discussing possible changes; a PR isn't the best place to discuss something with potentially significant behavioral changes where the correct outcome hasn't been decided.

Reply:
@stuartmorgan 
> I think the next step here is a [design document](https://flutter.dev/go/template) describing the difference and discussing possible changes; a PR isn't the best place to discuss something with potentially significant behavioral changes where the correct outcome hasn't been decided.

Thanks for the feedback, I will close this PR and start by submitting the design doc.